### PR TITLE
Remove optional new/const

### PR DESCRIPTION
**Pull request by natebosch**
- Run `dartfmt --fix`.
- Update version to start working towards next breaking change.
- Update minimum SDK to 2.0.0 stable.

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
 name: stream_channel
-version: 1.6.8
```
Comment (jakemac53) [inline pubspec.yaml]: given that there are not any breaking changes yet I would just do a dot release

also, please add a corresponding changelog entry

Comment (natebosch) [reply pubspec.yaml]: Done

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+## 1.6.9
+
+* Require `2.0.0` or newer SDK.
+* Drop unnecessary `new` and `const`.
```
Comment (kevmoo) [inline CHANGELOG.md]: This line can be dropped, right? Doesn't affect end users.

Comment (natebosch) [reply CHANGELOG.md]: 🤷‍♂️ Jake asked for an entry

https://github.com/dart-lang/stream_channel/pull/37#discussion_r220599243

Comment (kevmoo) [reply CHANGELOG.md]: @jakemac53 – you silly

**Pull request merged**